src/main/java/teammates/ui/controller/AdminSearchPageAction.java

```diff
@@ -266,17 +266,15 @@ private String findAvailableInstructorGoogleIdForCourse(String courseId){
         if(instructorList == null || instructorList.isEmpty()){
             return googleId;
         }
-        
-        for(InstructorAttributes instructor : instructorList){
-          
-            if(instructor.googleId != null){
-               googleId = instructor.googleId;
-               if(instructor.isAllowedForPrivilege(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)){            
-                   break;
-               } 
-            }            
+
+        for (InstructorAttributes instructor : instructorList) {
+
+            if (instructor.isRegistered() && instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)) {
```

Comment:
`INSTRUCTOR_PERMISSION_ROLE_COOWNER` is not a privilege name, so `isAllowedForPrivilege` is always false and the old code returns the last registered instructor in `instructorList`. Since #4777, it is guaranteed that there is always a registered instructor with the privilege to modify instructors in the course, therefore `INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR` is used.